modules/database/funcs.go

```diff
@@ -229,3 +231,51 @@ func (d *Database) getDBWithName(
 
 	return nil
 }
+
+func GetServiceDbByName(
+	ctx context.Context,
+	h *helper.Helper,
+	name string,
+	namespace string,
+) (*mariadbv1.MariaDBDatabase, error) {
+
+	db := &mariadbv1.MariaDBDatabase{}
+	err := h.GetClient().Get(ctx, types.NamespacedName{Name: name, Namespace: namespace}, db)
+	if err != nil {
+		if k8s_errors.IsNotFound(err) {
+			return nil, util.WrapErrorForObject(
+				fmt.Sprintf("Failed to get %s database %s ", name, h.GetBeforeObject().GetNamespace()),
+				h.GetBeforeObject(),
+				err,
+			)
+		}
+
+		return nil, util.WrapErrorForObject(
+			fmt.Sprintf("DB error %s %s ", name, h.GetBeforeObject().GetNamespace()),
+			h.GetBeforeObject(),
+			err,
+		)
+	}
+	return db, nil
+}
+
+func (d *Database) DeleteFinalizer(
```

Comment:
Does it compile with https://github.com/openstack-k8s-operators/placement-operator/pull/50/files#diff-c0f1c404ea10bc348fba06aab6f455b95b7513a337e09b325ebea615fc0ee587R211 ? It seems the placement patch calls this without the Database receiver.

Also I think you don't use `d` below so technically this does not need to be defined with a Database receiver.

Reply:
Reading the placement patch made me realize that you might have the Database receiver here so that you can call
```
db := NewDatabase()
db.DeleteFinalizer()
```
during the reconcileDelete() code path.

If that is the plan, then I would encourage to store the name and namespace of the MariaDBDatabase in the Database struct too so that the caller does not need to pass it to DeleteFinalizer any more.

Reply:
@stuggi @gibizer 
A little confused about my options here:
So I'm going for the logic of :

In lib common/funcs return an initialized  *database by using :
func GetServiceDbByName(
  ctx context.Context,
  h *helper.Helper,
  name string,
  namespace string,
) (*Database, error) {
  **db := NewDatabase()**
[..]
  return db, nil

Following that , I can in  DeleteFinalizer do this :
func (d *Database) DeleteFinalizer(
  ctx context.Context,
  h *helper.Helper,
  name string,
  namespace string,
) error {
  serviceDb, err := GetServiceDbByName(ctx, h, name, namespace)
  #not sure ^^ this belongs here or in the placementapi_controller
[..]
controllerutil.RemoveFinalizer(serviceDb, helper.GetFinalizer())

Then in  placementapi_controller.go I can instead of :
  err := database.DeleteFinalizer(ctx, helper, placement.ServiceName,
instance.Namespace)

do :

  db := GetServiceDbByName()
  db.DeleteFinalizer()

So that's one option .
Thing is I'm not sure how to pass the namesace and service name to the db.DeleteFinalizer() without passing them as parameters
like : GetServiceDbByName(ctx, h, name, namespace)
@gibizer  you said adding them to the *Database struct so something like:
..if GetServiceDbByName() returns the *Database..add those fields to the struct:
db := db{
    name: name,
    namespace: namespace,
}
then in the DeleteFinalizer() access the name and namespace like : d.name and d.namespace
or supply the d *Database object as:
controllerutil.RemoveFinalizer(d, helper.GetFinalizer())

Hope you get my code path confusion here..:)

Reply:
I think I like the following approach:
lib-common:
```golang
func GetDatabaseByName(
	ctx context.Context,
	h *helper.Helper,
	name string,
	namespace string,
) (*Database, error) {
    // you can create a Database here by only filling what we need (name and namespace)
    db := &Database{
		name:         name,
		namespace:    namespace,
    }
    // the querying the MariaDBDatabase and store it in db by calling
    db.getDBWithName(ctx, h)
    return db
```

```golang
func (d *Database) DeleteFinalizer(
    ctx context.Context,
    h *helper.Helper,
) error {
    controllerutil.RemoveFinalizer(d.database, helper.GetFinalizer())
```

placement:
```golang
db := GetDatabaseByName(ctx, helper, <dbname>, instance.Namespace)
db.DeleteFinalizer()
```

Reply:
thanks!
I'll run some checks , see how that compiles

Reply:
@gibizer  ,ok  with the latest changes you mentioned and a few patches I made seems to work at for the keystoneAPI : 
```sh
oc get MariaDBDatabase -o json|jq .items[].metadata.finalizers
[
  "KeystoneAPI"
]
[
  "PlacementAPI"
]
[r@r install_yamls]$ oc delete KeystoneAPI keystone
keystoneapi.keystone.openstack.org "keystone" deleted
[r@r install_yamls]$  oc get MariaDBDatabase -o json|jq .items[].metadata.finalizers
[
  "PlacementAPI"
]

```
thanks